src/bsp/src/stm32/src/interloc/src/states/src/SendResponseState.cpp

```diff
@@ -0,0 +1,37 @@
+#include <interloc/Decawave.h>
+#include <interloc/InterlocBSPContainer.h>
+#include <states/InterlocStateContainer.h>
+#include <states/SendResponseState.h>
+
+SendResponseState::SendResponseState(ILogger& logger, DecawaveArray& decawaves) :
+    AbstractInterlocState(logger, decawaves) {}
+
+void SendResponseState::process(InterlocStateHandler& context) {
+    // TODO: Set destination ID
+    context.constructUWBHeader(0x01, UWBMessages::DATA, UWBMessages::TWR_RESPONSE,
+                               (uint8_t*)&m_respMsg, sizeof(m_respMsg));
+    m_respMsg.m_subFrameId = RESPONDER_SLOT;
+
+    uint64_t respTxTime =
+        context.getTWR().m_pollRxTs + (POLL_RX_TO_RESP_TX_DLY_UUS * UUS_TO_DWT_TIME);
+    // deca.transmitAndReceive((uint8_t*)&m_respMsg, sizeof(m_respMsg), 0, m_rxFrame, 0);
+    m_decawaves[DecawavePort::A].transmitDelayedAndReceive(
```

Comment:
as pointed out previously, the wait_final state could exist due to the relatively long delay following last response transmission

Reply:
To change in the next task when we create the real states